ipam/provider/aws.go

```diff
@@ -0,0 +1,256 @@
+			CommonDomain:              getValue(config.CommonDomain, "COMMON_DOMAIN"),
+			SSHPublicKey:              getValue(config.SSHPublicKey, "IDRSA_PUB"),
+			VersionBundleVersion:      getValue(config.VersionBundleVersion, "VERSION_BUNDLE_VERSION"),
+		},
+		hostClusterName: getValue(config.ClusterName, "CLUSTER_NAME"),
+	}
+
+	return a, nil
+}
+
+func (aws *AWS) CreateCluster(clusterName string) error {
+	if clusterName == "" {
+		return microerror.Maskf(invalidConfigError, "clusterName must not be empty")
+	}
+
+	hostAWSConfig, err := aws.hostFramework.AWSCluster(aws.hostClusterName)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = aws.installCertResources(clusterName)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	deploymentName := awsConfigDeploymentName(clusterName)
+
+	o := func() error {
+		err := framework.HelmCmd(fmt.Sprintf("delete --purge %s", deploymentName))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		f, err := ioutil.TempFile("/tmp", deploymentName)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		defer func() {
+			// Close & remove created tempfile.
+			fName := f.Name()
+			err = f.Close()
+			if err != nil {
+				// XXX: Tempfile cannot be removed due to error. It must be
+				// removed manually.
+				aws.logger.Log("level", "error", "message", fmt.Sprintf("failed to close & remove tempfile '%s' - it must be removed manually", fName))
+				return
+			}
+			err = os.Remove(fName)
+			if err != nil {
+				// XXX: Tempfile cannot be removed due to error. It must be
+				// removed manually.
+				aws.logger.Log("level", "error", "message", fmt.Sprintf("failed to close & remove tempfile '%s' - it must be removed manually", fName))
+				return
+			}
+		}()
+
+		{
+			c := chartvalues.APIExtensionsAWSConfigE2EConfig{
+				CommonDomain:         aws.chartValuesConfig.CommonDomain,
+				ClusterName:          clusterName,
+				SSHPublicKey:         aws.chartValuesConfig.SSHPublicKey,
+				VersionBundleVersion: aws.chartValuesConfig.VersionBundleVersion,
+
+				AWS: chartvalues.APIExtensionsAWSConfigE2EConfigAWS{
+					Region:            aws.chartValuesConfig.AWSRegion,
+					APIHostedZone:     aws.chartValuesConfig.AWSAPIHostedZoneGuest,
+					IngressHostedZone: aws.chartValuesConfig.AWSIngressHostedZoneGuest,
+					RouteTable0:       clusterName + "_0",
+					RouteTable1:       clusterName + "_1",
+					VPCPeerID:         hostAWSConfig.Spec.AWS.VPC.PeerID,
+				},
+			}
+
+			values, err := chartvalues.NewAPIExtensionsAWSConfigE2E(c)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			_, err = f.WriteString(values)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+
+		err = framework.HelmCmd(fmt.Sprintf("registry install quay.io/giantswarm/apiextensions-aws-config-e2e-chart:stable -- -n %s --values %s", awsConfigDeploymentName(clusterName), f.Name()))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		return nil
+	}
```

Comment:
I think this can be simplified. Please check https://github.com/giantswarm/aws-operator/blob/d5994596e696081fbb693e45e6e3585004853df9/integration/setup/setup.go#L137-L170

Reply:
Good shout. I will.

Reply:
Uff.. That `err = config.Host.InstallResource("apiextensions-aws-config-e2e", values, ":stable")` uses hard coded deployment name. I would need to refactor that and fix existing e2e tests elsewhere. I would rather not do that at this point. If you don't mind, I'd rather leave this as-is as otherwise this rabbit hole never ends. WDYT?

Reply:
Similar to certconfig installation, I see this as part of the task when refactoring towards use of HelmClient instead of executing commands. I just had a chat with @MarcelMue and implementation for helm client is actually in the works currently and it seems good idea to make changes there for the API in order to incorporate these requirements and then make the switch. TODO for that is here: https://github.com/giantswarm/giantswarm/issues/4091